extractor/image/daemon/image.go

```diff
@@ -0,0 +1,134 @@
+package daemon
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"sync"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"golang.org/x/xerrors"
+)
+
+var mu sync.Mutex
```

Comment:
I'm curious why do we need a mutex? What call sites does this package have?

Reply:
`LayerByDiffID` is called by some goroutines at the same time to reduce the download time. If we don't have `sync.Mutex`, `populateImage` will be called many times. It means the same image is written down in a local file system many times.

Reply:
Got it.

Reply:
I got it. Could you add a little comment for this?

Reply:
Roger. I added https://github.com/aquasecurity/fanal/pull/96/commits/3833f9cf20d4a262151440f267761599b7a7019d